### PR TITLE
NET-3181 - Allow log file naming like Nomad

### PR DESCRIPTION
**Pull request by absolutelightning**
### Description
Fixes - https://github.com/hashicorp/consul/issues/15709
Currently consul logs files like this consul-{timestamp}.log
This change will create log files like this - 
```
asheshvidyut@asheshvidyut-H2GX766V9T /var/log » ls | grep consul
consul-1693400777323096000.log
consul-1693400777323902000.log
consul-1693400777324364000.log
consul-1693400777324796000.log
consul-1693400777325255000.log
consul-1693400777325694000.log
consul-1693400777326075000.log
consul-1693400777326486000.log
consul-1693400777326847000.log
consul-1693400777327154000.log
consul.log
asheshvidyut@asheshvidyut-H2GX766V9T /var/log »
```
So basically there will always be a `consul.log` with the latest logs in it.

### Testing & Reproduction steps

`make dev`
`sudo ./bin/consul agent -dev -log-rotate-max-files 10 -log-file /var/log/ -log-rotate-bytes 50`
Open browser and hit `localhost:8500` couple of times. This will create log files and I verified the content. 
`consul.log` had the latest logs as depicted by the timestamps of logs.

### PR Checklist

* [x] updated test coverage
* [x] external facing docs updated
* [x] appropriate backport labels added
* [x] not a security concern

### CODE REVIEW

**Review by aahel — APPROVED** [review]

File: logging/logfile.go [hunk 1]
```diff
@@ -79,12 +77,28 @@ func (l *LogFile) openNew() error {
 	return nil
 }
 
+func (l *LogFile) renameCurrentFile() error {
```
Comment (loshz) [inline logging/logfile.go:80]: Maybe we should add a test for this?

**Review by loshz — COMMENTED** [review]
The logic LGTM. Am I right in thinking this change means we always write to `consul.log` and when we hit the max bytes, we rename to `consul-{timestamp}.log`?

**Comment by absolutelightning** [comment]
> The logic LGTM. Am I right in thinking this change means we always write to `consul.log` and when we hit the max bytes, we rename to `consul-{timestamp}.log`?

yes. right

File: logging/logfile.go (hunk 1 above)
Comment (absolutelightning) [reply logging/logfile.go:80]: added

**Review by loshz — APPROVED** [review]

**Comment by david-yu** [comment]
Please follow up with backports when you're available. Thanks.

**Comment by jkirschner-hashicorp** [comment]
@absolutelightning : Did we confirm that `consul.log` is not overwritten when the agent starts up (if it already exists)?

When Nomad implemented this new logging behavior, they accidentally introduced such a bug. Their fix was in this PR: https://github.com/hashicorp/nomad/pull/11386.

**Comment by johnnyplaydrums** [comment]
Very excited for this update @absolutelightning, thank you for implementing! The docs still need to be updated - does that happen elsewhere or was it missed in this PR? I think [Consul's log file docs](https://developer.hashicorp.com/consul/docs/agent/config/cli-flags#_log_file) should be updated to more closely resemble[Nomad's log file docs](https://developer.hashicorp.com/nomad/docs/configuration#log_file). Thanks y'all.

**Pull request merged**